src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+
+import java.util.LinkedList;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private LinkedList<Entry<K, V>>[] table;
+    private int size;
+
+    private static class Entry<K, V> {
+        K key;
+        V value;
+
+        public Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    public MyHashMap() {
+        table = new LinkedList[INITIAL_CAPACITY];
+        size = 0;
+    }
+
+    // Метод для расчета индекса ячейки на основе хэш-кода ключа
+    private int hash(K key) {
+        return Math.abs(key.hashCode()) % table.length;
```

Comment:
The `hash` method does not handle `null` keys. If `key` is `null`, `key.hashCode()` will throw a `NullPointerException`. Consider adding a check for `null` keys and handling them appropriately.